projects/wvr-elements/src/lib/core/config.service.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConfigService {
+  baseUrl: string;
+
+  constructor(private http: HttpClient) {
+
+  }
+
+  load(): Promise<void> {
+    const headers = new HttpHeaders({ 'Content-Type': 'application/json' });
+    const url = 'config.json';
+
+    const promise = this.http.get(url, { headers })
+      .toPromise()
+      .then(configs => {
```

Comment:
Shouldn’t this `then` be on the promise variable assigned and not assign the results of the`then` and `catch` to the promise variable to be returned. I think callers of this function might be getting a different promise than expected.